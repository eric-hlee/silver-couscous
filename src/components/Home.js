import React from "react";
import { FormattedMessage } from "react-intl";
import Button from "react-bootstrap/Button";
import Col from "react-bootstrap/Col";
import Container from "react-bootstrap/Container";
import Image from "react-bootstrap/Image";
import Jumbotron from "react-bootstrap/Jumbotron";
import Row from "react-bootstrap/Row";
import About from "./About";
import Footer from "./Footer";

export default function Home() {
	return (
		<React.Fragment>
			<main>
				<header>
					<Jumbotron>
						<h1>
							<FormattedMessage id="home.greeting" defaultMessage="Hello!" />
						</h1>
						<p className="white">
							<FormattedMessage
								id="home.whoami"
								defaultMessage="My name is Eric."
							/>
						</p>
						<p>
							<Button
								variant="primary"
								href="http://localhost:9000/resume"
								className="icon-download"
							>
								<FormattedMessage id="home.resume" defaultMessage="Resume" />
							</Button>
						</p>
					</Jumbotron>
				</header>
				<section>
					<Container fluid>
						<Row>
							<Col className="img-eric">
								<Image
									src={require("../assets/images/eric.jpg")}
									alt="Picture of Eric"
									height="180"
									width="171"
									fluid
								/>
							</Col>
							<Col>
								<div className="speech-bubble">
									<p>
										<strong>
											<FormattedMessage
												id="home.life_as_dev"
												defaultMessage="🧑🏻‍💻 My life as a developer:"
											/>
										</strong>
									</p>
									<ul>
										<li>
											<FormattedMessage
												id="home.life_as_dev_1"
												defaultMessage="Started learning HTML at 10"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_2"
												defaultMessage="Received an award for creating a website at 12"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_3"
												defaultMessage="Studied computer science in university at 18"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_4"
												defaultMessage="Completed an internship at 22"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_5"
												defaultMessage="Kicked off career as a technical support engineer at 23"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_6"
												defaultMessage="Joined a startup as a junior software engineer at 24"
											/>
										</li>
										<li>
											<FormattedMessage
												id="home.life_as_dev_7"
												defaultMessage="Got laid off due to Covid-19 and looking for new opportunities!"
											/>
										</li>
									</ul>
								</div>
							</Col>
						</Row>
					</Container>
				</section>
				<About />
			</main>
			<Footer />
		</React.Fragment>
	);
}
