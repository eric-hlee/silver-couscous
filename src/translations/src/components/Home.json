[{"id":"home.greeting","defaultMessage":"Hello!","file":"src/components/Home.js","start":{"line":19,"column":7},"end":{"line":19,"column":70}},{"id":"home.whoami","defaultMessage":"My name is Eric.","file":"src/components/Home.js","start":{"line":22,"column":7},"end":{"line":25,"column":9}},{"id":"home.resume","defaultMessage":"Resume","file":"src/components/Home.js","start":{"line":33,"column":8},"end":{"line":33,"column":69}},{"id":"home.life_as_dev","defaultMessage":"🧑🏻‍💻 My life as a developer:","file":"src/components/Home.js","start":{"line":54,"column":11},"end":{"line":57,"column":13}},{"id":"home.life_as_dev_1","defaultMessage":"Started learning HTML at 10","file":"src/components/Home.js","start":{"line":62,"column":11},"end":{"line":65,"column":13}},{"id":"home.life_as_dev_2","defaultMessage":"Received an award for creating a website at 12","file":"src/components/Home.js","start":{"line":68,"column":11},"end":{"line":71,"column":13}},{"id":"home.life_as_dev_3","defaultMessage":"Studied computer science in university at 18","file":"src/components/Home.js","start":{"line":74,"column":11},"end":{"line":77,"column":13}},{"id":"home.life_as_dev_4","defaultMessage":"Completed an internship at 22","file":"src/components/Home.js","start":{"line":80,"column":11},"end":{"line":83,"column":13}},{"id":"home.life_as_dev_5","defaultMessage":"Kicked off career as a technical support engineer at 23","file":"src/components/Home.js","start":{"line":86,"column":11},"end":{"line":89,"column":13}},{"id":"home.life_as_dev_6","defaultMessage":"Joined a startup as a junior software engineer at 24","file":"src/components/Home.js","start":{"line":92,"column":11},"end":{"line":95,"column":13}},{"id":"home.life_as_dev_7","defaultMessage":"Got laid off due to Covid-19 and looking for new opportunities!","file":"src/components/Home.js","start":{"line":98,"column":11},"end":{"line":101,"column":13}}]
