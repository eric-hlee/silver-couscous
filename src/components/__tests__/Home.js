import React from "react";
import { shallow, mount } from "enzyme";
import { IntlProvider } from "react-intl";
import Home from "../Home";

describe("<Home />", () => {
	it("renders correctly", () => {
		const wrapper = shallow(<Home />);
		expect(wrapper).toMatchSnapshot();
	});

	it("calls /resume when download resume button is clicked", () => {
		const wrapper = mount(
			<IntlProvider locale="en">
				<Home />
			</IntlProvider>
		);
		wrapper.find("Button").simulate("click");
		global.window = Object.create(window);
		const href = "http://localhost:9000/resume";
		Object.defineProperty(window, "location", {
			value: {
				href,
			},
		});
		expect(window.location.href).toEqual(href);
		delete window.location;
	});
});
