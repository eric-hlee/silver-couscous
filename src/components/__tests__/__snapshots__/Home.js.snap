// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<Fragment>
  <main>
    <header>
      <Jumbotron
        fluid={false}
      >
        <h1>
          <FormattedMessage
            defaultMessage="Hello!"
            id="home.greeting"
            values={Object {}}
          />
        </h1>
        <p
          className="white"
        >
          <FormattedMessage
            defaultMessage="My name is Eric."
            id="home.whoami"
            values={Object {}}
          />
        </p>
        <p>
          <Button
            active={false}
            className="icon-download"
            disabled={false}
            href="http://localhost:9000/resume"
            type="button"
            variant="primary"
          >
            <FormattedMessage
              defaultMessage="Resume"
              id="home.resume"
              values={Object {}}
            />
          </Button>
        </p>
      </Jumbotron>
    </header>
    <section>
      <Container
        fluid={true}
      >
        <Row
          noGutters={false}
        >
          <Col
            className="img-eric"
          >
            <Image
              alt="Picture of Eric"
              fluid={true}
              height="180"
              rounded={false}
              roundedCircle={false}
              src="test-file-stub"
              thumbnail={false}
              width="171"
            />
          </Col>
          <Col>
            <div
              className="speech-bubble"
            >
              <p>
                <strong>
                  <FormattedMessage
                    defaultMessage="🧑🏻‍💻 My life as a developer:"
                    id="home.life_as_dev"
                    values={Object {}}
                  />
                </strong>
              </p>
              <ul>
                <li>
                  <FormattedMessage
                    defaultMessage="Started learning HTML at 10"
                    id="home.life_as_dev_1"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Received an award for creating a website at 12"
                    id="home.life_as_dev_2"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Studied computer science in university at 18"
                    id="home.life_as_dev_3"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Completed an internship at 22"
                    id="home.life_as_dev_4"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Kicked off career as a technical support engineer at 23"
                    id="home.life_as_dev_5"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Joined a startup as a junior software engineer at 24"
                    id="home.life_as_dev_6"
                    values={Object {}}
                  />
                </li>
                <li>
                  <FormattedMessage
                    defaultMessage="Got laid off due to Covid-19 and looking for new opportunities!"
                    id="home.life_as_dev_7"
                    values={Object {}}
                  />
                </li>
              </ul>
            </div>
          </Col>
        </Row>
      </Container>
    </section>
    <About />
  </main>
  <Footer />
</Fragment>
`;
